### PR TITLE
Add description for behavior changes of the DELETE REST operation caused by garabge collection

### PR DESCRIPTION
**Pull request by caesarxuchao**
@janetkuo @lavalamp @pwittrock

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33119)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -291,13 +292,17 @@ Old version kubectl's rolling-update command is compatible with Kubernetes 1.4 a
...
+If you use an old version kubectl to delete a replication controller or a replicaset, then after the delete command has returned, the replication controller or the replicaset will continue to exist in the key-value store for a short period of time (<1s). You probably will not notice any difference if you use kubectl manually, but you might notice it if you are using kubectl in a script. To fix it, you can poll the API server to confirm the object is deleted.
+
+### DELETE operation in REST API
+
+The DELETE request of a replication controller or a replicaset becomes asynchronous by default. The object will continue to exist in the key-value store for some time. The API server will set its metadata.deletionTimestamp, add the "orphan" finalizer to its metadata.finalizers. The object will be deleted from the key-value store after the garbage collector orphans its dependents. Please refer to this [user-guide](http://kubernetes.io/docs/user-guide/garbage-collector/) for more information regarding the garbage collection.
```
Comment (lavalamp) [inline CHANGELOG.md]: Not all objects get the orphan finalizer by default. in fact only RCs do? Please clarify.

Comment (caesarxuchao) [inline CHANGELOG.md]: I said it's only for RC and RS in the first sentence. Maybe I should emphasis is in the section title?

Comment (janetkuo) [inline CHANGELOG.md]: Yeah updating the title is a good idea

Comment (lavalamp) [inline CHANGELOG.md]: Yes, or add bullet points e.g.
* RC & RS:
 * text...
* other objects
 * no changes unless you explicitly request orphaning

Comment (caesarxuchao) [inline CHANGELOG.md]: Thanks. I took @lavalamp's suggestion. PTAL.

Comment (lavalamp) [inline CHANGELOG.md]: This can be wordsmithed more, but the content seems good.

**Pull request merged**